internal/cluster/connection_manager.go

```diff
@@ -265,6 +265,17 @@ func (m *ConnectionManager) checkClusterIDChanged() {
 	}
 }
 
+// GetClusterID returns the id of the connected cluster at that moment.
+func (m *ConnectionManager) GetClusterID() types.UUID {
+	m.clusterIDMu.Lock()
+	defer m.clusterIDMu.Unlock()
+	clusterID := m.clusterID
+	if clusterID == nil {
+		return types.UUID{}
```

Comment:
This looks like an erroneous case. If this indeed describes a fault, what adding a boolean to return values to indicate an error is observed?

Reply:
This is not an erroneous case. `clusterID` is set to `nil` At `ConnectionManager.reset`.

Reply:
Returning an empty UUID for the inconsistent case of `ClusterID` does more make sense to me. Therefore, I didn't change the current state.

Reply:
@nevzatseferoglu Could you update this method's documentation to reflect the current behavior of returning  `types.UUID{}` if the client is not connected to a cluster?

Reply:
Fixed at [ca6ca5c](https://github.com/hazelcast/hazelcast-go-client/pull/739/commits/ca6ca5c1ee38fffe75305bb1268ead9fe0c0e882).